### PR TITLE
Force update when local branch is ahead

### PR DESCRIPTION
**Pull request by fernandog**
@pymedusa/developers 

we recently had to remove one commit from develop which caused a "Local branch is ahead of develop. Automatic update not possible" to develop users

this will affect only develop and master branches.

### CODE REVIEW

File: sickbeard/versionChecker.py [hunk 1]
```diff
@@ -597,8 +598,7 @@ def set_newest_text(self):
 
             newest_text = 'There is a <a href="' + url + '" onclick="window.open(this.href); return false;">newer version available</a> '
             newest_text += " (you're " + str(self._num_commits_behind) + " commit"
-            if self._num_commits_behind > 1:
-                newest_text += 's'
```
Comment (fernandog) [inline sickbeard/versionChecker.py]: we already check if `self._num_commits_behind` is > so this will always be true

Comment (medariox) [inline sickbeard/versionChecker.py]: This adds an 's' to 'commit' if we are more than 1 commit behind.

Comment (fernandog) [inline sickbeard/versionChecker.py]: ahh I see now

**Comment by ratoaq2** [comment]
Is there a reason to have it only for master and develop?

**Comment by fernandog** [comment]
@ratoaq2 if someone is using a testing branch or working on another branch and haven't pushed yet, he wil lose the commits if this force updated it not only for develop|master

**Comment by ratoaq2** [comment]
Valid scenario. But there's another valid one:

- 1 'production' installation running a specific branch other than master or develop (for testing purposes, most of the time)
- 1 local dev environment

The local dev environment should never reset and force update, but the 'production' installation should.

File: sickbeard/versionChecker.py [hunk 2]
```diff
@@ -625,6 +626,12 @@ def need_update(self):
             if self._num_commits_behind > 0:
                 return True
 
+            # Force update if GIT_RESET enabled and commits ahead
+            if self._num_commits_ahead > 0 and sickbeard.GIT_RESET \
+               and sickbeard.BRANCH in ['develop', 'master']:
```
Comment (medariox) [inline sickbeard/versionChecker.py]: I'd keep this as one line, since it's < 120 chars.

**Comment by fernandog** [comment]
@ratoaq2 how to implement your scenario then?

File: gui/slick/views/config_general.mako [hunk 3]
```diff
@@ -730,7 +730,7 @@
                                 </span>
                             </label>
                         </div>
-                        <div class="field-pair" hidden>
+                        <div class="field-pair">
                             <label for="git_reset">
```
Comment (fernandog) [inline gui/slick/views/config_general.mako]: git_reset option was hidden

**Comment by fernandog** [comment]
@ratoaq2 can you test please?

File: sickbeard/versionChecker.py [hunk 4]
```diff
@@ -583,11 +583,7 @@ def set_newest_text(self):
         # if we're up to date then don't set this
         sickbeard.NEWEST_VERSION_STRING = None
 
-        if self._num_commits_ahead:
-            logger.log(u"Local branch is ahead of " + self.branch + ". Automatic update not possible.", logger.WARNING)
-            newest_text = "Local branch is ahead of " + self.branch + ". Automatic update not possible."
-
-        elif self._num_commits_behind > 0:
+        if self._num_commits_behind > 0 or (sickbeard.GIT_RESET and sickbeard.BRANCH in sickbeard.GIT_RESET_BRANCHES):
```
Comment (ratoaq2) [inline sickbeard/versionChecker.py]: If there's no branches in the list, it should accept as well

```
or (sickbeard.GIT_RESET and (not sickbeard.GIT_RESET_BRANCHES or sickbeard.BRANCH in sickbeard.GIT_RESET_BRANCHES)) 
```

The same logic some lines below (probably a function would fit: def should_reset)

Comment (fernandog) [inline sickbeard/versionChecker.py]: Did you try the Ui? Use must select branches in a pre-populated  select

**Comment by ratoaq2** [comment]
Approved

**Pull request merged**